src/orca-jedi/nemo_io/WriteServer.cc

```diff
@@ -0,0 +1,269 @@
+/// \param var_name   Name of the variable in the file.
+/// \param t_index    Time index in the file to write.
+/// \param buffer     vector of data to write.
+template<class T> void WriteServer::write_vol_var_on_root(const std::string& var_name,
+    const size_t t_index,
+    const std::vector<T>& buffer) {
+  oops::Log::trace() << "State(ORCA)::nemo_io::WriteServer::write_vol_var_on_root "
+                     << std::endl;
+  if (myrank == mpiroot) {
+    writer_->write_vol_var<T>(var_name, buffer, t_index);
+  }
+}
+template void WriteServer::write_vol_var_on_root<double>(const std::string& var_name,
+    const size_t t_index, const std::vector<double>& buffer);
+template void WriteServer::write_vol_var_on_root<float>(const std::string& var_name,
+    const size_t t_index, const std::vector<float>& buffer);
+
+/// \brief  Write a 2D field at a given time index to a file on the root process.
+/// \param var_name   Name of the variable in the file.
+/// \param t_index    Time index in the file to write.
+/// \param buffer     vector of data to write.
+template<class T> void WriteServer::write_surf_var_on_root(const std::string& var_name,
+    const size_t t_index,
+    const std::vector<T>& buffer) {
+  oops::Log::trace() << "State(ORCA)::nemo_io::WriteServer::write_surf_var_on_root "
+                     << std::endl;
+  if (myrank == mpiroot) {
+    writer_->write_surf_var<T>(var_name, buffer, t_index);
+  }
+}
+template void WriteServer::write_surf_var_on_root<double>(const std::string& var_name,
+    const size_t t_index, const std::vector<double>& buffer);
+template void WriteServer::write_surf_var_on_root<float>(const std::string& var_name,
+    const size_t t_index, const std::vector<float>& buffer);
+
+/// \brief sort the data into the correct order for the NetCDF file
+/// \param  buffer        .
+/// \return sorted_buffer vector of data to write.
+template<class T> std::vector<T> WriteServer::sort_buffer(
+  const std::vector<T> & buffer) const {
+  oops::Log::trace() << "State(ORCA)::nemo_io::WriteServer::sort_buffer "
+                     << std::endl;
+  std::vector<T> sorted_buffer(orca_indices_.nx() * orca_indices_.ny());
+  for (size_t i_buf = 0; i_buf < buffer.size(); ++i_buf) {
+    const size_t source_index = unsorted_buffer_indices_[i_buf];
+    ASSERT(source_index < sorted_buffer.size());
+    sorted_buffer[source_index] = buffer[i_buf];
+  }
+  return sorted_buffer;
+}
+template std::vector<double> WriteServer::sort_buffer<double>(
+    const std::vector<double> & buffer) const;
+template std::vector<float> WriteServer::sort_buffer<float>(
+    const std::vector<float> & buffer) const;
+
+/// \brief Gather field_view data from all processes to a buffer on the server root process
+/// \param field_view Field data to write to the file.
+/// \param i_level    level index for the data.
+/// \param start      start index for the destination buffer.
```

Comment:
This header doesn't look like it matches the arguments below.

Reply:
thanks for spotting this, I have updated it and it should be correct now.